server/chat-plugins/datasearch.ts

```diff
@@ -1043,6 +1035,7 @@ function runDexsearch(target: string, cmd: string, canAll: boolean, message: str
 	}
 	let results: string[] = [];
 	for (const mon of Object.keys(dex).sort()) {
+		if (singleTypeSearch && Object.keys(dex[mon].types).length !== 1) continue;
```

Comment:
Oh, wait, you shouldn't be using this.

```suggestion
		if (singleTypeSearch && dex[mon].types.length !== 1) continue;
```

Reply:
Alright, I'll open another PR for that.